common/validate.go

```diff
@@ -76,6 +76,22 @@ func ValidateDenom(denom string) error {
 	return nil
 }
 
+func ValidateDenomWithReserved(reservedDenoms []string) func(denom string) error {
+	return func(denom string) error {
+		for _, reserved := range reservedDenoms {
+			if denom == reserved {
+				return fmt.Errorf("denom %s is reserved and cannot be used", denom)
+			}
+		}
+
+		if !reDnm.MatchString(denom) {
```

Comment:
We should validate the denom format first before checking if it's in the reserved list. This provides better error handling since an invalid format should be caught before checking reserved status, and it's also more efficient as we avoid checking the reserved list for invalid denoms.